player/lua/ytdl_hook.lua

```diff
@@ -57,6 +59,35 @@ local function edl_escape(url)
     return "%" .. string.len(url) .. "%" .. url
 end
 
+local function time_to_secs(time_string)
+    if not time_string then
+        return nil
+    end
+
+    local secs = 0
+    string.gsub(time_string, "[%d.]+", function (number)
+        secs = secs * 60 + tonumber(number)
+    end)
+
+    return secs
+end
+
+local function extract_chapters(data, video_length)
+    local ret = {}
+
+    for line in data:gmatch("[^\r\n]+") do
+        local time_string = string.match(line, "([%d:.]+:[%d.]+)")
```

Comment:
You can't do optional capture groups, at least with plain lua, afaict.

How about:
```lua
function istime(str)
    local ret = nil
    local first, last = str:find("%d?%d:[0-5]%d")
    if first == nil then
    elseif first > 2 then
        if str:sub(first-1, first) == ":" then
            ret = str:match("(%d+:"..str:sub(first, last)..")")
        end
    elseif string.len(str) > last+2 then
        if str:sub(last+1, last+1) == ":" then
            ret = str:match("(%d*"..str:sub(first, last)..":[0-5]%d)")
        end
    else
        ret = str:sub(first, last)
    end
    if ret == nil then ret = "" end
    return ret
end
```

Result:
```bash
$ luajit-2.0.4 exp.lua
             2 =>
            :2 =>
           1:2 =>
          1:23 => 1:23
         01:23 => 01:23
        :01:23 => 01:23
        412:34 => 12:34
       1:23:35 => 1:23:35
      12:34:56 => 12:34:56
      12:34:65 =>
     123:45:45 => 123:45:45
       012:345 => 12:34
      0:12:345 => 0:12:34
     0:12:34.5 => 0:12:34
     0:123:456 =>
  da 0:123:456 =>
    01:23:45.6 => 01:23:45
   012:345:678 =>
  0:12:34:56.7 => 0:12:34
 0:12:34:56:78 => 0:12:34
0:12:34:56.789 => 0:12:34
```